api/features/workflows/core/models.py

```diff
@@ -69,19 +74,34 @@ def commit(self, committed_by: "FFAdminUser"):
                 "Change request has not been approved by all required approvers."
             )
 
-        feature_states = list(self.feature_states.all())
+        feature_states = list(
+            self.feature_states.all().select_related(
+                "environment", "environment__project"
+            )
+        )
 
         for feature_state in feature_states:
             if not feature_state.live_from:
                 feature_state.live_from = timezone.now()
-
             feature_state.version = FeatureState.get_next_version_number(
                 environment_id=feature_state.environment_id,
                 feature_id=feature_state.feature_id,
                 feature_segment_id=feature_state.feature_segment_id,
                 identity_id=feature_state.identity_id,
             )
 
+            # TODO: optimise these creates to use bulk create again but for now, we need to
+            # ensure the post_save signals on the AuditLog model class are triggered
+            message = FEATURE_STATE_UPDATED_MESSAGE % feature_state.feature.name
+            AuditLog.objects.create(
+                author=committed_by,
+                project=feature_state.environment.project,
+                environment=feature_state.environment,
+                related_object_type=RelatedObjectType.FEATURE_STATE.name,
```

Comment:
Thinking about it - this isn't necessarily correct, right? In the case that the Change Request is scheduled for the future. I guess we have a few options: 

 1. Create a different message / related object type for the audit log
 2. Only create the audit log if the feature states aren't scheduled for the future and we can write the audit log in the task when the feature states are released by the task processor

I am definitely leaning towards (1) here since this is an AuditLog, so we should definitely track when a Change Request is committed.

Reply:
I think this is resolved now from our previous discussions.